### PR TITLE
SPSH-893 Update email in itslearning

### PR DESCRIPTION
**Pull request by marode-cap**
# Description

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts [hunk 1]
```diff
@@ -36,60 +42,86 @@ export class ItsLearningPersonsEventHandler {
...
+    public async oxUserChangedEventHandler(event: OxUserChangedEvent): Promise<void> {
+        await this.personUpdateMutex.runExclusive(async () => {
+            this.logger.info(`Received OxUserChangedEvent, ${event.personId}`);
+
+            await this.updateEmail(event.personId, event.primaryEmail);
+        });
+    }
 
-        this.logger.info(`Person with ID ${event.personId} updated in itsLearning!`);
+    @EventHandler(OxMetadataInKeycloakChangedEvent)
```
Comment (phaelcg) [inline src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts:89]: Is it not the same as @EventHandler(OxUserChangedEvent)? When a user changes in ox, the user ifno will be updated in KC? Or does ox have another kc instance :)?

**Review by phaelcg — CHANGES_REQUESTED** [review]

File: src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts (hunk 1 above)
Comment (marode-cap) [reply src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts:89]: When I implemented this I asked @DPDS93CT if I need both, but I'll check if it is actually needed

File: src/modules/itslearning/actions/read-person.action.spec.ts [hunk 2]
```diff
@@ -53,11 +55,79 @@ describe('ReadPersonAction', () => {
...
+            const institutionRole: IMSESInstitutionRoleType = faker.helpers.enumValue(IMSESInstitutionRoleType);
+            const primaryRoleType: boolean = faker.datatype.boolean();
+            const action: ReadPersonAction = new ReadPersonAction(userId);
+
+            expect(
+                action.parseBody({
+                    readPersonResponse: {
+                        person: {
+                            name: {
+                                partName: [{ namePartType: 'First', namePartValue: firstName }],
```
[the remark this answers is not in the document]
Comment (marode-cap) [reply src/modules/itslearning/actions/read-person.action.spec.ts:81]: Done

File: src/modules/itslearning/actions/read-person.action.spec.ts [hunk 3]
```diff
@@ -53,11 +55,79 @@ describe('ReadPersonAction', () => {
...
+            const institutionRole: IMSESInstitutionRoleType = faker.helpers.enumValue(IMSESInstitutionRoleType);
+            const primaryRoleType: boolean = faker.datatype.boolean();
+            const action: ReadPersonAction = new ReadPersonAction(userId);
+
+            expect(
+                action.parseBody({
+                    readPersonResponse: {
+                        person: {
+                            name: {
+                                partName: [{ namePartType: 'Last', namePartValue: lastName }],
```
[the remark this answers is not in the document]
Comment (marode-cap) [reply src/modules/itslearning/actions/read-person.action.spec.ts:114]: Done

File: src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts (hunk 1 above)
Comment (marode-cap) [reply src/modules/itslearning/event-handlers/itslearning-persons.event-handler.ts]: `OxUserChangedEvent` seems to be the only relevant event, the other one is thrown inside of an `OxUserChangedEvent`-handler. I've removed the second event-handler so we just have the `OxUserChangedEvent`-handler

**Review by phaelcg — APPROVED** [review]
Do not forget the playwright tests :)

**Pull request merged**